app/src/main/java/de/westnordost/streetcomplete/quests/max_weight/AddMaxWeightForm.kt

```diff
@@ -0,0 +1,175 @@
+
+            override fun onNothingSelected(parentView: AdapterView<*>) {}
+        }
+
+        /* Workaround for an Android bug that it assumes the decimal separator to always be the "."
+           for EditTexts with inputType "numberDecimal", independent of Locale. See
+           https://issuetracker.google.com/issues/36907764 .
+
+           Affected Android versions are all versions till (exclusive) Android Oreo. */
+
+        /* actually, let's not care about which separator the user uses, he might be confused
+           whether he should use the one as displayed on the sign or in his phone's locale */
+        //char separator = DecimalFormatSymbols.getInstance(getCountryInfo().getLocale()).getDecimalSeparator();
+        tonInput?.keyListener = DigitsKeyListener.getInstance("0123456789,.")
+
+        switchLayout(unit)
+    }
+
+    private fun switchLayout(unit: Measurement) {
+        val isMetric = unit == METRIC
+        val isImperial = unit == IMPERIAL
+
+        tonInputSign?.visibility = if (isMetric) View.VISIBLE else View.GONE
+        poundInputSign?.visibility = if (isImperial) View.VISIBLE else View.GONE
+
+        if (isMetric) tonInput?.requestFocus()
+        if (isImperial) poundInput?.requestFocus()
+    }
+
+    private fun getSpinnerItems(units: List<String>) = units.mapNotNull {
+        when(it) {
+            "metric" -> "t"
+            "imperial" -> "lbs"
+            else -> null
+        }
+    }
+
+    override fun onClickOk() {
+        if (userSelectedUnrealisticHeight()) {
+            confirmUnusualInput { applyMaxWeightFormAnswer() }
+        } else {
+            applyMaxWeightFormAnswer()
+        }
+    }
+
+    private fun userSelectedUnrealisticHeight(): Boolean {
+        val weight = getWeightFromInput() ?: return false
+        val w = weight.toTons()
+        return w > 25 || w < 2
+    }
+
+    private fun applyMaxWeightFormAnswer() {
+        applyAnswer(MaxWeight(getWeightFromInput()!!))
+    }
+
+    private fun getWeightFromInput(): Measure? {
+        try {
+            if (isMetric()) {
+                val input = InputUtil.numberInputToStandardString(tonInput)
+                if (input.isNotEmpty()) {
+                    val country = countryInfo.countryCode
+                    if ("US" == country) {
+                        return UsShortTons(input.toDouble())
+                    }
+                    return MetricMeasure(input.toDouble())
+                }
+            } else {
+                val poundString = InputUtil.numberInputToStandardString(poundInput)
+
+                if (poundString.isNotEmpty()) {
+                    return ImperialMeasure(poundString.toInt())
+                }
+            }
+        } catch (e: NumberFormatException) {
+            return null;
+        }
+        return null
+    }
+
+    private fun isMetric() =
+        weightUnitSelect?.let { it.selectedItem == "t" }
+            ?: (countryInfo.measurementSystem[0] == "metric")
```

Comment:
What about short tons?

Reply:
US sign have short tons described as "t" so it is also displayed this way and later converted to st for US locations.